msal/application.py

```diff
@@ -309,6 +346,7 @@ def get_authorization_request_url(
             self.http_client
             ) if authority else self.authority
 
+        claims = _merge_claims_and_capabilities(self.client_capabilities, claims)
```

Comment:
Assigning a different value into the same input parameter is error-prone. I would suggest to either use a different variable name:

```suggestion
        _claims = _merge_claims_and_capabilities(self.client_capabilities, claims)
```

or see the comment below.